qiskit/opflow/evolutions/trotterizations/qdrift.py

```diff
@@ -42,26 +42,33 @@ def __init__(self, reps: int = 1) -> None:
         super().__init__(reps=reps)
 
     def convert(self, operator: OperatorBase) -> OperatorBase:
-        # TODO: implement direct way
+        if not isinstance(operator, (SummedOp, PauliSumOp)):
+            raise TypeError("Trotterization converters can only convert SummedOp or PauliSumOp.")
+
         if isinstance(operator, PauliSumOp):
-            operator = operator.to_pauli_op()
-
-        if not isinstance(operator, SummedOp):
-            raise TypeError('Trotterization converters can only convert SummedOps.')
-
-        summed_op = cast(SummedOp, operator)
-        # We artificially make the weights positive, TODO check approximation performance
-        weights = np.abs([op.coeff for op in summed_op.oplist])  # type: ignore
-        lambd = sum(weights)
-        N = 2 * (lambd ** 2) * (summed_op.coeff ** 2)
-
-        factor = lambd * summed_op.coeff / (N * self.reps)
-        # The protocol calls for the removal of the individual coefficients,
-        # and multiplication by a constant factor.
-        scaled_ops = \
-            [(op * (factor / op.coeff)).exp_i() for op in summed_op.oplist]  # type: ignore
-        sampled_ops = aqua_globals.random.choice(scaled_ops,
-                                                 size=(int(N * self.reps),),  # type: ignore
-                                                 p=weights / lambd)
+            # We artificially make the weights positive, TODO check approximation performance
+            weights = np.abs(operator.primitive.coeffs)
+            lambd = np.sum(weights)
+
+            N = 2 * (lambd ** 2) * (operator.coeff ** 2)
+            factor = lambd * operator.coeff / (N * self.reps)
+            # The protocol calls for the removal of the individual coefficients,
+            # and multiplication by a constant factor.
+            scaled_ops = [(op * (factor / op.coeff)).exp_i() for op in operator]
+
+        if isinstance(operator, SummedOp):
+            # We artificially make the weights positive, TODO check approximation performance
+            weights = np.abs([op.coeff for op in operator.oplist])
+            lambd = sum(weights)
+            N = 2 * (lambd ** 2) * (operator.coeff ** 2)
+
+            factor = lambd * operator.coeff / (N * self.reps)
+            # The protocol calls for the removal of the individual coefficients,
+            # and multiplication by a constant factor.
+            scaled_ops = [(op * (factor / op.coeff)).exp_i() for op in operator.oplist]
```

Comment:
I think the two impls for PauliSumOp and SummedOp are almost identical. How about unifying them as follows?
```python
            # We artificially make the weights positive, TODO: check approximation performance
            weights = np.abs([op.coeff for op in operator.oplist])
            lambd = sum(weights)
            N = 2 * (lambd ** 2) * (operator.coeff ** 2)

            factor = lambd * operator.coeff / (N * self.reps)
            # The protocol calls for the removal of the individual coefficients,
            # and multiplication by a constant factor.
            operator_list = operator if isinstance(operator, PauliSumOp) else operator.oplist
            scaled_ops = [(op * (factor / op.coeff)).exp_i() for op in operator_list]
```